logger.go

```diff
@@ -250,6 +250,21 @@ func (log *Logger) Core() zapcore.Core {
 	return log.core
 }
 
+func (log *Logger) SetCore(core zapcore.Core) {
+	log.core = core
```

Comment:
This is racy, since there may be concurrent calls to the logger that are accessing the core.

in general, I don't think mutability of the underlying core is something we should do